test/e2e/network_test.go

```diff
@@ -827,31 +828,34 @@ func (tc *testContext) deleteJob(name string) error {
 	return jobsClient.Delete(context.TODO(), name, metav1.DeleteOptions{})
 }
 
-// waitUntilJobSucceeds will return an error if the job fails or reaches a timeout
-func (tc *testContext) waitUntilJobSucceeds(name string) error {
+// waitUntilJobSucceedsAndFetchLogs will return an error if the job fails or reaches a timeout and return logs on success
+func (tc *testContext) waitUntilJobSucceedsAndFetchLogs(name string) (string, error) {
```

Comment:
I feel this name change is overly verbose. Its a common pattern for functions like this to return logs, having a note in the function doc as you did should be good enough.